### ISSUE TITLE
Add `Setter Inflation Strategy` similar to staking pallet's rewards inflation strategy 

### ISSUE THREAD

**Issue by balqaasem** [setheum/setheum#303]
Add SettCurrency Inflation to SerpTreasury and inflation will be issued to the `setter_reward_pool` which will then hold the inflation of setter, some amount is issued per block to make it up to an average of around 0.258% to 2.58% annually.
* Do it through a Welfare pallet - namely "WelfareTreasury"
The setter_reward_pool will have 4 recipients of it's entire funds:
```
1. SettPay Treasury (for cashdrops) = 50%,
2. Charity Fund = 30%,
3. WelfareTreasury to buy back and burn DNARNEOM (through DEX-Swap only, not auctions) = 10%,
4. Setheum Treasury = 10%,
```
The SERP will call the `payout_setter_rewards` at every `SetterRewardPayoutTime` (this could be anything between 10 minutes and 1 hour).

NOTE:- Only the Setter "SETT/NSETT" is eligible for this, it is Setheum's de-facto stablecoin.

GOVERNANCE: And handle the params to the `MonetaryCouncil`, to let them set these params:
`update_max_welfare_inflation_rate(); // this lets the council approve an update on the "MaxWelfareInflationRate"`
`update_min_welfare_inflation_rate(); // this lets the council approve an update on the "MinWelfareInflationRate"`
These governance parameters make an elastic monetary policy possible for the Setter and the Setheum Economy, making it yet the most advanced economy I know.

**Comment by balqaasem** [comment]
Name the pallet `setheum_welfare_treasury` - `setheum-welfare-treasury` - `WelfareTreasury`

**Comment by balqaasem** [comment]
Only buy back DNAR/NEOM through DEX (SettinDex/HalalSwap)

**Comment by balqaasem** [comment]
No more dedicated to the initially designated beneficiaries I listed above.. They will be issued to stakers in the staking pallet.

**Comment by balqaasem** [comment]
Closes in #368
